barretenberg/cpp/src/barretenberg/vm2/tracegen/alu_trace.cpp

```diff
@@ -30,13 +31,51 @@ std::vector<std::pair<Column, FF>> get_operation_columns(const simulation::AluEv
                  // I think the only situation in which a + b != c as fields is when c overflows the bit size
                  // if this in unclear, I can use > or actually check bit sizes:
                  { Column::alu_cf, event.a.as_ff() + event.b.as_ff() != event.c.as_ff() } };
+    case simulation::AluOperation::MUL: {
+        bool is_u128 = event.a.get_tag() == ValueTag::U128;
+        uint256_t a_int = static_cast<uint256_t>(event.a.as_ff());
+        uint256_t b_int = static_cast<uint256_t>(event.b.as_ff());
+        // Columns shared for all tags in a MUL:
+        std::vector<std::pair<Column, FF>> res = {
+            { Column::alu_sel_op_mul, 1 },
+            { Column::alu_op_id, SUBTRACE_INFO_MAP.at(ExecutionOpCode::MUL).subtrace_operation_id },
+            { Column::alu_constant_64, 64 },
+            { Column::alu_sel_is_u128, is_u128 },
+            { Column::alu_tag_u128_diff_inv,
+              is_u128
+                  ? 0
+                  : FF((static_cast<uint8_t>(event.a.get_tag())) - static_cast<uint8_t>(MemoryTag::U128)).invert() },
```

Comment:
This needs to be FF(a_tag) - FF(U128_tag). Instead of doing a uint8_t subtraction and then casting to the field